components/date-picker/RangePicker.tsx

```diff
@@ -62,13 +64,31 @@ function fixLocale(value: RangePickerValue | undefined, localeCode: string) {
   }
 }
 
-export default class RangePicker extends React.Component<any, RangePickerState> {
+class RangePicker extends React.Component<any, RangePickerState> {
   static defaultProps = {
     prefixCls: 'ant-calendar',
     allowClear: true,
     showToday: false,
   };
 
+  static getDerivedStateFromProps(nextProps: any, prevState: any) {
+    let state = null;
+    if (('value' in nextProps) && !isEqual(nextProps.value,prevState.value)) {
+      const value = nextProps.value || [];
+      state = {
+        value,
+        showDate: getShowDateFromValue(value) || prevState.showDate,
+      };
+    }
+    if (('open' in nextProps) && prevState.open !== nextProps.open) {
```

Comment:
this doesn't solve the problem:  `state.open` still got reset to `props.open` whenever `getDerivedStateFromProps` is called.

One possible solution is to store previous `props.open` & `props.value` in state and compare them in gDSFP, then update state if they didn't match with `nextProps`

Reply:
@whtsky that's what I am doing no ? previous values of 'open' and 'value' are stored in state and I am comparing them in gDSFP ??

Reply:
oh, sorry. Didn’t realize value & open won’t be modified if passed in props.